### ISSUE TITLE
Operator toList invalidates process cache

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#207]
When `toList` is applied to a channel of values it will invalidates the cache of the process to which it is applied. For example: 

```
process foo {
  input: 
  val x from 1,2,3
  output: 
  val x into channel 

  """
  echo $x
  """
}

process bar {
  input: 
  val x from channel.toList()

  """
  echo $x
  """

}
```

When _resuming_ the execution of this script the `bar` process may not use the cached result because the execution order of `foo` process may be changed.

**Comment by pditommaso** [comment]
using the following input declaration in the `bar` process, solves the problem 

```
   val x from channel.toList().map{ new nextflow.util.ArrayBag(it) } 
```

**Comment by pditommaso** [comment]
Resolved by #297.
